File: src/buildings/smith.js
import * as PIXI from 'pixi.js'
import { Building } from '../building.js'

export class Smith extends Building {

    constructor (game) {
        super("Smith",{x:2,y:2},[[0,0],[0,1]],game,true)
        this.sprite = PIXI.Sprite.from('Smith')
        this.recipe = {
            Sandstone: 20
        }

        this.description = `A building that allows you to craft new items for your species.`
    }

    //recipes stored here 
    //need to create item object
}

File: TriloGame.CSharp/src/TriloGame.Game/Core/Buildings/Smith.cs
using TriloGame.Game.Core.Simulation;
using TriloGame.Game.Shared.Math;

namespace TriloGame.Game.Core.Buildings;

public sealed class Smith : Building
{
    public Smith(GameSession session)
        : base("Smith", new GridPoint(2, 2), [[0, 0], [0, 1]], session, true)
    {
        TextureKey = "Smith";
        Recipe = new Dictionary<string, int>(StringComparer.Ordinal) { ["Sandstone"] = 20 };
        Description = "A building that allows you to craft new items for your species.";
    }
}
